app/furniture/marketplace/cart.rb

```diff
@@ -54,7 +54,7 @@ def price_total
     end
 
     def ready_for_checkout?
-      delivery.details_filled_in? && cart_products.present? && cart_products.all?(&:valid?)
+      delivery_area.present? && cart_products.present? && cart_products.all?(&:valid?)
```

Comment:
I was curious about where this method used but when I scanned the codebase I found no invocations/references?

Reply:
It was not being used, which was why this bug exists. Removing the constraint that it requires address information and adding the call to `ready_for_checkout?` back to the `Checkout` button in [`carts/_footer.html.erb`](https://github.com/zinc-collective/convene/pull/1991/files#diff-7669466e6284058bbba9590f22f1a03283f91386f361314995a1824369349d38R15) is what fixes the bug.

Reply:
Yeah I see the fix, I was confused about why it wasn't already invoked somewhere and thought maybe something funny was going on. All looks good though.

Reply:
I think it was when we were moving entering the `Delivery Address` to the `Checkout` page the `ready_for_checkout?` would never be true until too late. A smarter fix would have been to change it to checking the `delivery_area` but  ah well.